src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -100,12 +100,20 @@ public void storeMessage(Message message) {
     // people may include implicit context/reference to previously sent messages,
     // and this order makes the chronology of the conversation and context
     // more intuitive and understandable in the messages list page. 
-    Query query =
+    Query query;
```

Comment:
I see two issues here:

   1. Both branches of the `if` are identical. Therefore, the branch is not needed.
   2. Even if conditional logic were necessary, a declaration without an assignment statement is generally a code smell; when conditional initialization logic is necessary, use a subroutine and/or ternary to always put an assignment in the declaration statement. As in:

```
Query query = hasUserRestriction(loggedInUser, otherUser)
                        ? createMessageQueryWithUserFilter(loggedinUser, otherUser)
                        : createPublicMessageQuery();
```

Reply:
I did not know that someone's code could even smell. TIL